asserts/checkers.go

```diff
@@ -38,12 +39,18 @@ func checkMandatory(headers map[string]string, name string) (string, error) {
 	return value, nil
 }
 
-func checkAssertType(assertType AssertionType) (*assertionTypeRegistration, error) {
-	reg := typeRegistry[assertType]
-	if reg == nil {
-		return nil, fmt.Errorf("unknown assertion type: %v", assertType)
+var errInvalidAssertionType = errors.New("invalid assertion type")
```

Comment:
@niemeyer Assemble uses Type() so can check directly, this is more use in places that actually get a *AssertionType